### PR TITLE
[screensaver] Avoid getting stuck in dim state when skin is reloaded

### PR DESCRIPTION
**Pull request by popcornmix**

### CODE REVIEW

**Comment by popcornmix** [comment]
This bug has been around for years. See:
http://forum.kodi.tv/showthread.php?tid=236524
http://forum.kodi.tv/showthread.php?tid=115688
http://forum.kodi.tv/showthread.php?tid=131883
https://discourse.osmc.tv/t/dim-screensaver-wont-un-dim/7126

Basically if screensaver is active and something causes a skin reload (perhaps a skin update) then the gui gets stuck in dimmed state even when woken up.

The issue is we get an UpdateVisibility(0) to disable the screensaver and we close CGUIDialog, but don't save the dim level. The dialogs get opened again and we don't get a subsequent UpdateVisibility(0), meaning screensaver is inactive, but still has the 80% dim level.

You can test this my switching skins when dim is active. e.g. with:
``./texturecache.py setsetting lookandfeel.skin skin.amber``

**Comment by MrMC** [comment]
nice, I've looked on and off for years trying to find that one.

**Comment by fritsch** [comment]
Thanks very much. We need a backport, right?

jenkins build this please

File: xbmc/windows/GUIWindowScreensaverDim.cpp [hunk 1]
```diff
@@ -40,11 +40,9 @@ CGUIWindowScreensaverDim::~CGUIWindowScreensaverDim(void)
 void CGUIWindowScreensaverDim::UpdateVisibility()
 {
   float level = g_application.GetDimScreenSaverLevel();
+  m_dimLevel = level;
```
Comment (mkortstiege) [inline xbmc/windows/GUIWindowScreensaverDim.cpp]: I'd make it `m_dimLevel = g_application.GetDimScreenSaverLevel(); if (m_dimlevel) ..`.
Really just a minor and more a personal taste. Thanks for checking and fixing that ugly issue!

**Comment by popcornmix** [comment]
Added the cosmetic suggested by @mkortstiege
jenkins build this please

**Comment by mkortstiege** [comment]
Thanks.

**Comment by MilhouseVH** [comment]
Minor nit:

With this PR, when the Dim screensaver is deactivated there's an instant transition from the dimmed state to undimmed state.

Prior to this PR, the transition after deactivating the Dim screensaver was more graceful, clearly stepping through several stages of dim, ie. 80% dim, 70% dim, 60% dim until eventually 0% dim is reached. Now, when the Dim screensaver is deactivated, it seems to instantly change from 80% dim to 0% dim without these intermediate levels.

The intermediate levels are still applied when the Dim screensaver is activated (though in reverse), which is why this new behaviour looks a little odd.

**Comment by MilhouseVH** [comment]
Confirming https://github.com/popcornmix/xbmc/commit/e1a3bd3884a4f3639fa3127e711d954c63a5f2e3 restores the traditional undim behaviour - thanks @popcornmix.

**Pull request merged**